src/CheckBox/index.jsx

```diff
@@ -36,8 +36,8 @@ class CheckBox extends React.Component {
         checked: true,
       });
     }
-    if (this.onChange) {
-      this.onChange();
+    if (this.onChangeState) {
+      this.onChangeState();
```

Comment:
So, this seems to be a little complicated. In the `asInput` wrapper, the `onChange` attribute is used instead of `onClick` like in the checkbox. It seems like `onChange` is not called when the checkbox is clicked. So, I’m not sure if we can get rid of the `handleClick` function. 

Further, `asInput`’s `onChange` is hit when the checkbox calls the`console.log(“the checkbox changed state”)` function. Hence why I had to change the name from `onChange` to `onChangeState` because when the `console.log(“the checkbox changed state”)` function is called, asInput’s `onChange` handler is called which then calls the passed in `onChange` function a second time and I get errors.